src/measure_theory/function/simple_func_dense_lp.lean

```diff
@@ -895,6 +895,60 @@ begin
   exact λ f hf, h_ae hf.coe_fn_to_Lp (Lp.mem_ℒp _) (this (hf.to_Lp f)),
 end
 
+/-- If a set of ae strongly measurable functions is stable under addition and approximates
+characteristic functions in `ℒp`, then it is dense in `ℒp`. -/
+lemma mem_ℒp.induction_dense (hp_ne_top : p ≠ ∞) (h'p : 1 ≤ p) (P : (α → E) → Prop)
```

Comment:
I find it easier to read statements like this when written in terms of filters. I guess this is equivalent to the following:
```lean
lemma mem_ℒp.induction_dense'  (hp_ne_top : p ≠ ∞) [fact (1 ≤ p)] (P : Lp E p μ → Prop)
  (h0P : ∀ (c : E) ⦃s⦄ (hs : measurable_set s) (hs' : μ s ≠ ∞),
    ∃ᶠ g in 𝓝 (indicator_const_Lp p hs hs' c), P g)
  (h1P : ∀ f g, P f → P g → P (f + g))
  (h2P : ∀ f, P f → ae_strongly_measurable f μ) (f : Lp E p μ) :
  ∃ᶠ g in 𝓝 f, P g :=
```

I don't have much experience with this area of the codebase and I can imagine it is more desirable to work in terms of raw functions (rather than terms of `Lp`) but perhaps it is worth adding this version also?

Reply:
Similar remarks apply to some of the other lemmas in this PR.

Reply:
The new version you are proposing is essentially a direct consequence of `Lp.induction`. The point of the lemma is indeed to work with genuine functions, avoiding all reference to the topology of `L^p` as going back and forth between genuine functions and equivalence classes is quite painful. The same goes for the rest of the PR: in the file `continuous_map_dense`, it is hard to express the fact intrinsically the fact that compactly supported functions are dense in `L^p` because it would require building several additional gadgets, while the statement as given with epsilons is straightforward.

So, after thinking about it I'll rather merge the current version, and we can revisit later if needed.

Reply:
(FWIW, your message crossed with one I was writing expressing the same view – for "approximation by nice functions" problems it is awkward to work directly with elements of `Lp`, because mathlib's preferred representative of the ae-eq class of a nice function won't usually be itself nice.)